test/core/consensus/grandpa/vote_tracker_test.cpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "consensus/grandpa/impl/vote_tracker_impl.hpp"
+#include "consensus/grandpa/structs.hpp"
+
+#include <gtest/gtest.h>
+#include "testutil/literals.hpp"
+
+using namespace kagome::consensus::grandpa;
+using kagome::common::Hash256;
+using PushResult = VoteTracker<Prevote>::PushResult;
+
```

Comment:
You can utilize [type-parameterized tests](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#type-parameterized-tests) and test against Precommits as well